docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -0,0 +1,463 @@
+return, other than having new peers joining their network.
+
+From an operational point of view, the design of the p2p layer should take into
+consideration the fact that not every node in the network is a "good" state
+sync server. Fresh nodes, which need support for bootstrapping, should then be
+able to discover peers in the network that are able or willing to provide
+application snapshots.
+The implementation of this feature would require a more complex content-based
+peer discovery mechanism.
+
+However, while relevant, strengthening p2p-based state sync should be seen as
+_orthogonal_ to the solution proposed in this ADR, which should have relevant use
+cases even for an enhanced version of state sync.
+
+### Manual application state transfer
+
+The proposed [ADR 083][adr083]/[ADR 103][adr103] inspired the discussions that
+led to this ADR. At first glance, their proposal might look identical to the
+solution discussed here. The main distinction is that these older proposals do not
+involve installing application _snapshots_, but rely on node operators to
+manually synchronize the application _state_.
+
+More precisely, their proposal is to
+"_allow applications to start with a bootstrapped state_
+(i.e., the application has already a state before the node is started)
+_alongside an empty CometBFT instance_
+(i.e., the node's block store is empty and the state store is at genesis state)
+_using a subsystem of the state sync protocol_".
+
+Starting from an "empty CometBFT instance" is a requirement for running state
+sync, which is maintained in the current proposal.
+The mentioned "subsystem of the state sync protocol" is the light client,
+responsible for verifying the application state and for bootstrapping
+CometBFT's block store and state store accordingly.
+
+The distinction, therefore, lies in the way which the state of the application
+from a running node is transferred to the new node to be bootstrapped.
+Instead of relying on application snapshots, produced by the application but
+handled by state sync, [ADR 083][adr083]/[ADR 103][adr103] assumes that node
+operators are able to manually synchronize the application state from a running
+node (it might be necessary to stop it) to a not-yet-started fresh node.
+
+The main limitation of the approach in [ADR 083][adr083]/[ADR 103][adr103] 
+is that it relies on the ability of node
+operators to properly synchronize the application state between two nodes.
+While experienced node operators are likely able to perform this operation in a
+proper way, we have to consider a broader set of users and emphasize that it is
+an operation susceptible to errors.
+Furthermore, it is an operation that is, by definition, application-specific:
+applications are free to manage their state as they see fit, and this includes
+how and where the state is persisted.
+Node operators would therefore need to know the specifics of each application
+in order to adopt this solution.
```

Comment:
Somewhere in this paragraph, I would also mention the risk of data corruption (e.g. in a manually transferred DB) going undetected with the manual process in ADR103

Reply:
I think the actual risk is that the oeprator has to read the applications DB, then dump the contenst, without knowing the exact format of the application's snapshot DB. But def something to add.